### ISSUE TITLE
Inmersive Mode

### ISSUE THREAD

**Issue by GuanacoDevs** [firebase/FirebaseUI-Android#1206]
This is more of a question than an issue.
Is it possible to set inmersive mode in FirebaseUI, I have checked on the library but don't find anything.

**Comment by SUPERCILEX** [comment]
No, we don't support this even though I doubt it would be too difficult. This is because Google, Facebook, and Twitter would all have to support immersive mode, which I'm pretty sure none of them do. @samtstern do you think we'll ever be able to properly support immersive mode?

**Comment by GuanacoDevs** [comment]
Not sure if a follow you properly, when you say about the Federated providers you mean the intent of each of them,? I was thinking mostly on the Activity that host the signing buttons for any provider, I can live with the non Immersive mode for the intents for any federated providers but having it only for the Screen with the login buttons would mix better with my app style.
Thought that adding this lines on `HelperActivityBase` could do it, which I call in `onResume()`:

```
    public void setImmersive() {
        int flags = View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                | View.SYSTEM_UI_FLAG_FULLSCREEN;
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
            flags = flags | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY;
        }
        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            getWindow().getDecorView().setSystemUiVisibility(flags);
            final View decorView = getWindow().getDecorView();
            final int finalFlags = flags;
            decorView.setOnSystemUiVisibilityChangeListener(new View.OnSystemUiVisibilityChangeListener() {
                @Override
                public void onSystemUiVisibilityChange(int visibility) {
                    if ((visibility & View.SYSTEM_UI_FLAG_FULLSCREEN) == 0) {
                        decorView.setSystemUiVisibility(finalFlags);
                    }
                }
            });
        }
    }
```
Thank you anyway @SUPERCILEX

**Comment by SUPERCILEX** [comment]
Oh I see, so you've forked it? And you don't mind if the external provider logins will quit immersive mode? Also, you shouldn't need to fork it if you just want AuthUI to be immersive, you can supply a [style to use everywhere](https://github.com/firebase/FirebaseUI-Android/blob/master/auth/README.md#themes) which should let you set those flags, right?

**Comment by GuanacoDevs** [comment]
@SUPERCILEX No, I don't mind the external providers quitting the immersive mode, Purchases already do, and I actually have set my custom theme like below, I still have the Navigation Bar at the bottom:

```
     <style name="login_theme" parent="FirebaseUI">
        <item name="colorPrimary">@color/colorPrimary</item>
        <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
        <item name="colorAccent">@color/colorAccent</item>
        <item name="colorButtonNormal">@color/colorAccent</item>
        <item name="colorControlNormal">@android:color/white</item>
        <item name="colorControlActivated">@android:color/white</item>
        <item name="colorControlHighlight">@android:color/white</item>
        <item name="android:textColor">@android:color/white</item>
        <item name="android:textColorPrimary">@android:color/white</item>
        <item name="android:textColorSecondary">@android:color/white</item>
        <item name="android:windowBackground">@drawable/gradient_main_background</item>
        <item name="windowActionBar">false</item>
        <item name="windowNoTitle">true</item>
        <item name="android:windowFullscreen">true</item>
        <item name="android:textColorHint">@android:color/white</item>
        <item name="android:immersive">true</item>
    </style>
```

**Comment by SUPERCILEX** [comment]
I've gotten it to work with this code: (I've got to say, it looks so damn sexy full screen 😆)
```java
public void setImmersive() {
    int flags = View.SYSTEM_UI_FLAG_LAYOUT_STABLE
            | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
            | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
            | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
            | View.SYSTEM_UI_FLAG_FULLSCREEN;

    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
        flags = flags | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY;
    }

    final View decorView = getWindow().getDecorView();
    decorView.setSystemUiVisibility(flags);

    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
        final int finalFlags = flags;
        decorView.setOnSystemUiVisibilityChangeListener(new View.OnSystemUiVisibilityChangeListener() {
            @Override
            public void onSystemUiVisibilityChange(int visibility) {
                decorView.setSystemUiVisibility(finalFlags);
            }
        });
    }
}
```

I don't think there's anything we can do in XML since there isn't a flag for the navigation bar hiding.

**Comment by GuanacoDevs** [comment]
@SUPERCILEX It does uh :-)
Oh well, but thanks to you I have found this:
`        <item name="android:navigationBarColor">@color/transparent</item>
`
I think I'll settle with that, is  `Build.VERSION_CODES.LOLLIPOP`, but looks good.
Tried also with:
`        <item name="android:windowTranslucentNavigation">true</item>
`
which is `Build.VERSION_CODES.LOLLIPOP`, But it draws the login buttons to the bottom.
Besides with the `NavigationBar` showing(transparent for me) it gives the user the option that they still can backout.
Thank you @SUPERCILEX , been very helpfull.
